l3build.dtx

```diff
@@ -1662,35 +1663,40 @@
 %
 % \begin{variable}{options}
 %   The |options| table holds the values passed to \pkg{l3build} at the
-%   command line. The possible entries in the table are given in the table
-%   below.
+%   command line. Each possible \meta{entry} given below corresponds to
+%   |--|\meta{entry} given at the command line, except |names| that
+%   corresponds to \meta{name(s)} for |check|, |doc| and |save| targets.  
 %   \begin{center}
 %   \begin{tabular}{ll}
 %     \toprule
 %     Entry & Type \\
 %     \midrule
-%       \var{config}        & Table   \\
-%       \var{date}          & String  \\
-%       \var{dirty}         & Boolean \\
-%       \var{dry-run}       & Boolean \\
-%       \var{email}         & String  \\
-%       \var{engine}        & Table   \\
-%       \var{epoch}         & String  \\
+%       \var{config}        & table   \\
+%       \var{date}          & string  \\
+%       \var{dirty}         & boolean \\
+%       \var{dry-run}       & boolean \\
+%       \var{email}         & string  \\
+%       \var{engine}        & table   \\
+%       \var{epoch}         & string  \\
 %       \var{file}          & string  \\
-%       \var{first}         & Boolean \\
-%       \var{full}          & Boolean \\
-%       \var{halt-on-error} & Boolean \\
-%       \var{help}          & Boolean \\
+%       \var{first}         & boolean \\
+%       \var{full}          & boolean \\
+%       \var{halt-on-error} & boolean \\
+%       \var{help}          & boolean \\
 %       \var{message}       & string  \\
-%       \var{names}         & Table   \\
-%       \var{quiet}         & Boolean \\
-%       \var{rerun}         & Boolean \\
-%       \var{shuffle}       & Boolean \\
-%       \var{stdengine}     & Boolean \\
-%       \var{texmfhome}     & String  \\
+%       \var{names}         & table \\
+%       \var{quiet}         & boolean \\
+%       \var{rerun}         & boolean \\
+%       \var{shuffle}       & boolean \\
+%       \var{stdengine}     & boolean \\
+%       \var{texmfhome}     & string  \\
 %     \bottomrule
 %     \end{tabular}
 %   \end{center}
+%   The Lua tables mentionned here are in fact arrays of strings.
+%   From the |build.lua| file, one can modify the string and boolean values,
+%   add or remove entries in arrays. But it is not recommended to affect a
+%   whole new Lua table to |options| nor to its array entries.
```

Comment:
I',m not sure what you are getting at: `options` is set by `l3build <options>`

Reply:
Apparently, options is not a read only variable. It is set by `l3buil-arguments` before `build.lua` is executed and modified afterwards. Nothing prevents people from saying
```
options={...}
```
or
```
options.engine = {...}
```
in `build.lua` or `config-....lua`. This is an open door to insidious bugs.